test/suites/dev/moonbase/test-eth-tx/test-eth-tx-access-list.ts

```diff
@@ -0,0 +1,183 @@
+import "@moonbeam-network/api-augment";
+import { beforeAll, deployCreateCompiledContract, describeSuite, expect } from "@moonwall/cli";
+import { createViemTransaction } from "@moonwall/util";
+import { error } from "node:console";
+
+describeSuite({
+  id: "D011304",
+  title: "Ethereum Transaction - Access List",
+  foundationMethods: "dev",
+  testCases: ({ context, it }) => {
+    let data;
+    let helper;
+    let helperProxy;
+
+    beforeAll(async () => {
+      helper = await deployCreateCompiledContract(context, "AccessListHelper");
+      helperProxy = await deployCreateCompiledContract(context, "AccessListHelperProxy", {
+        args: [helper.contractAddress],
+      });
+    });
+
+    it({
+      id: "T01",
+      title: "after the 4th one, additional storage keys should cost 1900 gas",
+      test: async function () {
+        const keys = generateSequentialStorageKeys(100);
+
+        interface Results {
+          keys: number;
+          size: number;
+          gasWithAL: bigint;
+        }
+
+        const cases = Array.from({ length: 100 }, (_, i) => i + 1);
+
+        const results: Results[] = [];
+
+        for (const n of cases) {
+          const txWithAL = await createViemTransaction(context, {
+            to: helperProxy.contractAddress,
+            data: data,
+            gas: 1000000n,
+            accessList: [
+              {
+                address: helper.contractAddress,
+                storageKeys: keys.slice(0, n),
+              },
+            ],
+          });
+
+          await context.createBlock(txWithAL);
+          const block = await context.viem().getBlock();
+          const receipt = await context
+            .viem()
+            .getTransactionReceipt({ hash: block.transactions[0] as `0x${string}` });
+          const gasCostWithAL = receipt.gasUsed;
+          const txSize = txWithAL.length;
+
+          results.push({
+            keys: n,
+            size: txSize,
+            gasWithAL: gasCostWithAL,
+          });
+        }
+
+        results.forEach((result, index) => {
+          const diff = index === 0 ? "" : result.gasWithAL - results[index - 1].gasWithAL;
+          if (result.keys > 4) {
+            expect(diff).toBe(1900n);
```

Comment:
Maybe add a message to the expect so if it fails we know which number it failed on.